float64 link_length_1
float64 link_length_2
float64 link_length_3
float64 link_length_4
float64 link_length_5
float64 link_length_6
float64 link_length_7

float64 x
float64 y
float64 z
---
float64 angle1
float64 angle2
float64 angle3
float64 angle4
float64 angle5
float64 angle6